src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -191,21 +187,183 @@ def construct_vnfd_parameters(self) -> Dict[str, Any]:
+        :type bicep_template_path: str
+        :parameters_json_file: path to an override file of set parameters for the nfdv
+        :param manifest_bicep_path: The path to the bicep template of the manifest
+        :param manifest_parameters_json_file: path to an override file of set parameters for
+                                        the manifest
+        """
+        assert isinstance(self.config, NSConfiguration)
+
+        if not bicep_path:
+            # User has not passed in a bicep template, so we are deploying the default
+            # one produced from building the NSDV using this CLI
+            bicep_path = os.path.join(
+                self.config.build_output_folder_name,
+                NSD_DEFINITION_BICEP_FILE,
+            )
+
+        if parameters_json_file:
+            message = f"Use parameters from file {parameters_json_file}"
+            logger.info(message)
+            print(message)
+            with open(parameters_json_file, "r", encoding="utf-8") as f:
+                parameters = json.loads(f.read())
+
+        else:
+            # User has not passed in parameters file, so we use the parameters required
+            # from config for the default bicep template produced from building the
+            # NSDV using this CLI
+            logger.debug("Create parameters for default NSDV template.")
+            parameters = self.construct_nsd_parameters()
+
+        logger.debug(parameters)
+
+        # Create or check required resources
+        deploy_manifest_template = not self.nsd_predeploy()
+
+        if deploy_manifest_template:
+            self.deploy_manifest_template(
+                manifest_parameters_json_file, manifest_bicep_path, NSD
+            )
+        else:
+            print(f"Artifact manifests {self.config.acr_manifest_name} already exists")
+
+        message = (
+            f"Deploy bicep template for NSDV {self.config.nsd_version} "
+            f"into {self.config.publisher_resource_group_name} under publisher "
+            f"{self.config.publisher_name}"
+        )
+        print(message)
+        logger.info(message)
+        self.deploy_bicep_template(bicep_path, parameters)
+        print(
+            f"Deployed NSD {self.config.acr_manifest_name} version {self.config.nsd_version}."
+        )
+        acr_manifest = ArtifactManifestOperator(
+            self.config,
+            self.api_clients,
+            self.config.acr_artifact_store_name,
+            self.config.acr_manifest_name,
+        )
+
+        arm_template_artifact = acr_manifest.artifacts[0]
+
+        ## Convert the NF bicep to ARM
+        arm_template_artifact_json = self.convert_bicep_to_arm(
+            os.path.join(self.config.build_output_folder_name, NF_DEFINITION_BICEP_FILE)
+        )
+
+        arm_template_artifact_path = os.path.join(
+            self.config.build_output_folder_name, NF_DEFINITION_JSON_FILE
```

Comment:
This is a schema which I'm guessing is the parameters file based on the bicep? If it is, I think we should rename it name.parameters.json, as this is how it would be names if you clicked "generate parameters file" in vs code and seems to be the standard

Reply:
As discussed, this is not a parameters file but an ARM template for the NF. I am just going to delete this file at the end of the `publish` step to not show unnecessary information for the user.